spec/service/integration/person_intergration_spec.rb

```diff
@@ -0,0 +1,337 @@
+        expect(last_response.status).to eq 404
+      end
+
+      it "returns the correct error message when attempting to update an invalid id" do
+        patch "/1/Person/#{id}",
+        {"name": "John"}.to_json,
+        { 'CONTENT_TYPE' => 'application/json; charset=utf-8' }
+        message = %({
+          "errors": [
+            {
+              "code": "generic.not_found",
+              "message": "Resource not found",
+              "reference": "#{id}"
+            }
+          ],
+          "kind": "Errors"})
+        expect(last_response.body).to be_json_eql(message).excluding("interaction_id")
+      end
+
+      it "returns the correct error message when attempting to update an invalid id" do
+        patch "/1/Person/#{person.id}",
+        {"name": "John", "number_of_cats": "12"}.to_json,
+        { 'CONTENT_TYPE' => 'application/json; charset=utf-8' }
+        message = %({
+          "errors": [
+            {
+              "code": "generic.invalid_parameters",
+              "message": "Body data contains unrecognised or prohibited fields",
+              "reference": "number_of_cats"
+            }
+          ],
+          "kind": "Errors"})
+        expect(last_response.body).to be_json_eql(message).excluding("interaction_id")
+      end
+    end
+  end
+
+
+  context "#delete" do
+
+    context "Person exists" do
+
+      let(:person){FactoryBot.create(:person)} # mock an instance of a person for testing
+
+      it "returns 200 if successful" do
+        delete "/1/Person/#{person.id}",
+        nil,
+        { 'CONTENT_TYPE' => 'application/json; charset=utf-8' }
+        expect(last_response.status).to eq 200
+      end
+
+      it "deletes a person correctly" do
+        delete "/1/Person/#{person.id}",
+        nil,
+        { 'CONTENT_TYPE' => 'application/json; charset=utf-8' }
+        message = %({"name": "#{person.name}","kind": "Person"})
+        expect(last_response.body).to be_json_eql(message)
+      end
+
+    end
+
+    context "Person doesn't exist" do
+      let(:id) { Hoodoo::UUID.generate}# generate a random UUID to mock an id that is not in the db.
+
+      it "returns 404 if unsucessful" do
+        delete "/1/Person/#{id}",
+        nil,
+        { 'CONTENT_TYPE' => 'application/json; charset=utf-8' }
+        expect(last_response.status).to eq 404
+      end
+
+      it "deletes a person correctly" do
+        delete "/1/Person/#{id}",
+        nil,
+        { 'CONTENT_TYPE' => 'application/json; charset=utf-8' }
+        msg = %(
+          {
+            "errors": [
+              {
+                "code": "generic.not_found",
+                "message": "Resource not found",
+                "reference": "#{id}"
+              }
+            ],
+            "kind": "Errors"
+          }
+        )
+        expect(last_response.body).to be_json_eql(msg).excluding("interaction_id")
+      end
+    end
+  end
+
+  context "#list" do
+
+    context "database contains multiple people" do
```

Comment:
This is great.   Can you add another contact that tests the `list` function works when the database is empty